graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/InterfaceGenerator.kt

```diff
@@ -20,20 +20,27 @@ package com.netflix.graphql.dgs.codegen.generators.java
 
 import com.netflix.graphql.dgs.codegen.CodeGenConfig
 import com.netflix.graphql.dgs.codegen.CodeGenResult
+import com.netflix.graphql.dgs.codegen.filterSkipped
+import com.netflix.graphql.dgs.codegen.shouldSkip
 import com.squareup.javapoet.ClassName
 import com.squareup.javapoet.JavaFile
 import com.squareup.javapoet.MethodSpec
 import com.squareup.javapoet.TypeSpec
 import graphql.language.*
 import javax.lang.model.element.Modifier
 
-class InterfaceGenerator(config: CodeGenConfig, private val document: Document) {
+class InterfaceGenerator(private val config: CodeGenConfig, private val document: Document) {
 
     private val packageName = config.packageNameTypes
     private val typeUtils = TypeUtils(packageName, config, document)
     private val useInterfaceType = config.generateInterfaces
 
     fun generate(definition: InterfaceTypeDefinition, extensions: List<InterfaceTypeExtensionDefinition>): CodeGenResult {
+
+        if (definition.shouldSkip(config)) {
+            return CodeGenResult()
```

Comment:
nit: the empty constructor as a stand-in for a code generation result which is empty is a bit implicit - it might help someone coming later to the project if it was explicitly identified either via a wrapping type or an explicit empty value initialized once